### PR TITLE
DaisySP integration

### PR DESCRIPTION
**Pull request by antisvin**
It generally works. Patch header should contain the following:

```
#include "Patch.h"

#undef min
#undef max
#undef abs
#undef sin
#undef cos
#undef exp
#undef sqrt
#undef pow
#undef log
#undef log10

#include "daisysp.h"

using namespace daisysp;
```

DaisySP objects can be statically or dynamically allocated, then constructor should call .Init(getSampleRate()) for each of them.

### CODE REVIEW

**Comment by pingdynasty** [comment]
Got any example patches?

**Comment by antisvin** [comment]
Sure, I'll share a patch or two today.

**Comment by antisvin** [comment]
Example1, should look vaguely familiar :-) - https://github.com/antisvin/MyPatches/blob/master/DaisySP/KickBoxDPatch.hpp

It would be too demanding for OWL2, so I'll add something simpler

**Comment by pingdynasty** [comment]
Will need at least one good example which shows clearly how to use it, and which works on all (or most) devices.
Also does the submodule need to point to your fork? What are the differences that can't be pushed upstream?

**Comment by antisvin** [comment]
Some news:

1. Resolved some issues with math, now it seems to run about twice as fast than before on F4
2. Fixing math (and integrating our optimizations) requires overriding their "dsp.h" header which we can't do easily as library is using includes with double quotes. This causes local file to be used with higher priority than anything we can specify as -I or alternatives
3. They've just released an update that completely changes modules layout, but at least files are group in directories that are easier to navigate. However this required extra updates to makefile for building everything - just solved recently
4. The fork is required until they merge [this](https://github.com/electro-smith/DaisySP/pull/138) or provide some other way for overriding dsp.h. If that won't happen, we can move this fork in your repo or add library source. 
5. Here's a more lightweight (but filthy!) patch [here](https://github.com/antisvin/MyPatches/blob/master/DaisySP/FilthyKickPatch.hpp), I use it for testing on Magus

File: compile.mk [hunk 1]
```diff
@@ -16,10 +16,21 @@ SOURCE       = $(BUILDROOT)/Source
...
+CPPFLAGS += -idirafter $(DAISYSP)
+CPPFLAGS += -idirafter $(DAISYSP)/Control
+CPPFLAGS += -idirafter $(DAISYSP)/Drums
+CPPFLAGS += -idirafter $(DAISYSP)/Dynamics
+CPPFLAGS += -idirafter $(DAISYSP)/Effects
+CPPFLAGS += -idirafter $(DAISYSP)/Filters
+CPPFLAGS += -idirafter $(DAISYSP)/Noise
+CPPFLAGS += -idirafter $(DAISYSP)/PhysicalModeling
+CPPFLAGS += -idirafter $(DAISYSP)/Synthesis
+CPPFLAGS += -idirafter $(DAISYSP)/Utility
```
Comment (pingdynasty) [inline compile.mk]: Why do you use -idirafter here?

**Review by pingdynasty — APPROVED** [review]

File: compile.mk (hunk 1 above)
Comment (antisvin) [reply compile.mk]: That's due to their excellent taste in naming headers ;-)
https://github.com/electro-smith/DaisySP/issues/137
So including it with -I causes wrong file getting used in basicmath.c and a compilation error due to memset not getting found there.

**Comment by antisvin** [comment]
Switched to use upstream repo as submodule, no problems with test patch

**Pull request merged**